src/lib/model/OutboundRequestToPayModel.js

```diff
@@ -0,0 +1,490 @@
+                    // }
+
+                    // cancel the timeout handler
+                    clearTimeout(timeout);
+
+                    // stop listening for transfer fulfil messages
+                    this._cache.unsubscribe(transactionRequestKey, subId).catch(e => {
+                        this._logger.log(`Error unsubscribing (in callback) ${transactionRequestKey} ${subId}: ${e.stack || util.inspect(e)}`);
+                    });
+
+                    if (error) {
+                        return reject(error);
+                    }
+
+                    const transactionRequestResponse = message.data;
+                    this._logger.push({ transactionRequestResponse }).log('Transaction Request Response received');
+                    this.data.requestToPayState = transactionRequestResponse.transactionRequestState;
+
+                    
+                    return resolve(transactionRequestResponse);
+                }
+                catch(err) {
+                    return reject(err);
+                }
+            });
+
+            // set up a timeout for the request
+            const timeout = setTimeout(() => {
+                const err = new BackendError(`Timeout waiting for fulfil for transfer ${this.data.transferId}`, 504);
+
+                // we dont really care if the unsubscribe fails but we should log it regardless
+                this._cache.unsubscribe(transactionRequestKey, subId).catch(e => {
+                    this._logger.log(`Error unsubscribing (in timeout handler) ${transactionRequestKey} ${subId}: ${e.stack || util.inspect(e)}`);
+                });
+
+                return reject(err);
+            }, this._requestProcessingTimeoutSeconds * 1000);
+
+            // now we have a timeout handler and a cache subscriber hooked up we can fire off
+            // a POST /transfers request to the switch
+            try {
+                const res = await this._requests.postTransactionRequests(transactionRequest, this.data.to.fspId);
+                this._logger.push({ res }).log('Transfer prepare sent to peer');
+            }
+            catch(err) {
+                // cancel the timout and unsubscribe before rejecting the promise
+                clearTimeout(timeout);
+
+                // we dont really care if the unsubscribe fails but we should log it regardless
+                this._cache.unsubscribe(transactionRequestKey, subId).catch(e => {
+                    this._logger.log(`Error unsubscribing (in error handler) ${transactionRequestKey} ${subId}: ${e.stack || util.inspect(e)}`);
+                });
+
+                return reject(err);
+            }
+        });
+    }
+
+    /**
+     * Builds a transaction request payload from current state
+     *
+     * @returns {object} - the transaction request payload
+     */
+    _buildTransactionRequest() {
+        let transactionRequest = {
+            transactionRequestId: this.data.transactionRequestId,
```

Comment:
@ksatya77 see my other comment on generating the transactionRequestId here rather than requiring DFSP backend to generate it. Please do same as we do with transferId in outbound transfers model.

Reply:
I now removed this as a required field in the incoming request.
// add a transactionRequestId if one is not present e.g. on first submission
        if(!this.data.hasOwnProperty('transactionRequestId')) {
            this.data.transactionRequestId = uuid();
        }
this adds the transactionRequestId and it is sent back in the response.